models/categories.models.js

```diff
@@ -5,3 +5,24 @@ exports.fetchCategories = () => {
     return categories;
   });
 };
+
+exports.fetchCategoryBySlug = (slug) => {
```

Comment:
okay so I added the promise all here to stop the case where a category exists but has no reviews from returning an error. So checking that the returned query is an empty array but also that the category doesn't exist in the list of all categories. Maybe there's a better way to do it than this?